eoflow/models/layers.py

```diff
@@ -1,338 +1,304 @@
+        return self.layer(inputs, training=training)
 
-        :param input_: Input 4D tensor of shape [N, H, W, C]
-        :param ksize: Size of pooling kernel along height and width of tensor. Default is `2`
-        :param stride: Stride of pooling along height and width of tensor. Default is `2`
-        :return: Tensor resulting form max pooling
-    """
-    return tf.nn.max_pool(input_,
-                          ksize=[1, ksize, ksize, 1],
-                          strides=[1, stride, stride, 1],
-                          padding='SAME')
-
-
-def max_pool_3d(input_, ksize=2, stride=2, pool_time=False):
-    """ Max pooling of a 5D tensor which represents either a 2D+t or 3D (multi-channel) input image
-
-        :param input_: Input 5D tensor of shape [N, D, H, W, C]. D could be time or depth of image.
-        :param ksize: Size of pooling kernel. If `pool_time` is `True`, the kernel size along D is `1`, otherwise is
-                        equal to ksize. Default is `2`
-        :param stride: Stride of pooling kernel. If `pool_time` is `True`, the kernel size along D is `1`, otherwise is
-                        equal to stride. Default is `2`
-        :param pool_time: Whether to operate pooling along the 2nd dimension (time or depth). Default is `False`
-        :return: Tensor resulting from max pooling
-    """
-    ktsize = ksize if pool_time else 1
-    tstride = stride if pool_time else 1
 
-    return tf.nn.max_pool3d(input_,
-                            ksize=[1, ktsize, ksize, ksize, 1],
-                            strides=[1, tstride, stride, stride, 1],
-                            padding='SAME')
+class Reduce3DTo2D(tf.keras.layers.Layer):
+    """ Reduces 3d representations into 2d using 3d convolution over the whole time dimension. """
+
+    def __init__(self, filters, kernel_size=3, stride=1, add_dropout=False, dropout_rate=0.2):
+        super().__init__()
+
+        self.filters = filters
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.add_dropout = add_dropout
+        self.dropout_rate = dropout_rate
+        self.layer = None
+
+    def build(self, input_size):
+        t_size = input_size[1]
+        layer = []
+        layer.append(tf.keras.layers.Conv3D(
+            self.filters,
+            kernel_size=(t_size, self.kernel_size, self.kernel_size),
+            strides=(1, self.stride, self.stride),
+            padding='VALID',
+            activation='relu'
+        ))
 
+        if self.add_dropout:
+            layer.append(tf.keras.layers.Dropout(rate=self.dropout_rate))
+
+        self.layer = tf.keras.Sequential(layer)
+
+    def call(self, inputs, training=None):
+        r = self.layer(inputs, training=training)
+
+        # Squeeze along temporal dimension
+        return tf.squeeze(r, axis=[1])
+
+
+# OLD, NOT YET CONVERTED
 
 def conv2d_gru(input_, nfeats_out, k_size=3, scope='reduce_t', padding='VALID', return_sequence=False):
```

Comment:
I think this should also be removed before merging. It's leftover code, not really useful without converting.